Makefile

```
@@ -184,17 +187,22 @@ eclint/lint:
 	$(ECLINT_FILES) | grep -zv ".bats" | xargs -0 -I {} eclint check {}
 	@ echo "[$@]: Project PASSED eclint test!"
 
-python/%: FIND_PYTHON := find . $(FIND_EXCLUDES) -name '*.py' -type f
-## Lints Python files
-python/lint: | guard/program/black
+python/%: PYTHON_FILES := git ls-files '*.py'
+## Checks format and lints Python files.  Runs pylint on each individual
+## file and uses a custom format for the lint messages.
+python/lint: | guard/program/pylint guard/program/black guard/program/git
+python/lint:
 	@ echo "[$@]: Linting Python files..."
-	$(FIND_PYTHON) | $(XARGS) black --check $$(dirname {})
+	$(PYTHON_FILES) | xargs black --check
+	$(PYTHON_FILES) | ${XARGS} -n1 pylint -rn -sn \
+		--msg-template="{path}:{line} [{symbol}] {msg}" {}
 	@ echo "[$@]: Python files PASSED lint test!"
 
-## Formats Python files
-python/format: | guard/program/black
+## Formats Python files.
+python/format: | guard/program/black guard/program/git
+python/format:
 	@ echo "[$@]: Formatting Python files..."
-	$(FIND_PYTHON) | $(XARGS) black $$(dirname {})
+	$(PYTHON_FILES) | xargs black
```

this will fix the test failure:


```suggestion
	$(PYTHON_FILES) | $(XARGS) black {}
```

---

No ... this causes black to be executed twice.  When xargs is used, it returns a 123, not the actual exit code:  

> 123 if any invocation of the command exited with status 1-125

I've changed the tests to look for 123.

---

Pretty sure the tests need to look for a status of 2. This is a bats thing, iirc, the actual exit code is not what it is checking.

---

It does need a status of 2.  I had added a line in the tests to echo the value of $status and saw 123.    Later I got 2.  I don't really understand what was going on at certain points.  For a long time the 'git ls-files ...' command was retrieving all the bats files in the directory in addition to the python files, which caused python/lint tests to fail.  I finally deleted the tardigrade-ci directory, checked out my files and started again.  That cleaned things up.  As you noted, the use of test git files by the bats test cases needs to be handled a better way.